src/Features/Core/Portable/ExtractClass/ExtractClassWithDialogCodeAction.cs

```diff
@@ -0,0 +1,264 @@
+                    extractClassOptions.MemberAnalysisResults.Select(m => m.Member),
+                    _document.Project.Solution,
+                    originalTypeDeclaration,
+                    cancellationToken).ConfigureAwait(false);
+
+                var fileBanner = syntaxFacts.GetFileBanner(root);
+                var namespaceService = _document.GetRequiredLanguageService<AbstractExtractInterfaceService>();
+
+                // Create the symbol for the new type 
+                var newType = CodeGenerationSymbolFactory.CreateNamedTypeSymbol(
+                    _selectedType.GetAttributes(),
+                    _selectedType.DeclaredAccessibility,
+                    _selectedType.GetSymbolModifiers(),
+                    TypeKind.Class,
+                    extractClassOptions.TypeName);
+
+                var containingNamespaceDisplay = namespaceService.GetContainingNamespaceDisplay(
+                    _selectedType,
+                    _document.Project.CompilationOptions);
+
+                // Add the new type to the solution. It can go in a new file or
+                // be added to an existing. The returned document is always the document
+                // containing the new type
+                var (updatedDocument, typeAnnotation) = extractClassOptions.SameFile
+                    ? await ExtractTypeHelpers.AddTypeToExistingFileAsync(
+                        symbolMapping.AnnotatedSolution.GetRequiredDocument(_document.Id),
+                        newType,
+                        symbolMapping,
+                        cancellationToken).ConfigureAwait(false)
+                    : await ExtractTypeHelpers.AddTypeToNewFileAsync(
+                        symbolMapping.AnnotatedSolution,
+                        containingNamespaceDisplay,
+                        extractClassOptions.FileName,
+                        _document.Project.Id,
+                        _document.Folders,
+                        newType,
+                        fileBanner,
+                        cancellationToken).ConfigureAwait(false);
+
+                // Update the original type to have the new base
+                var solutionWithUpdatedOriginalType = await GetSolutionWithBaseAddedAsync(
+                    updatedDocument.Project.Solution,
+                    symbolMapping,
+                    newType,
+                    extractClassOptions.MemberAnalysisResults,
+                    cancellationToken).ConfigureAwait(false);
+
+                // After all the changes, make sure we're using the most up to date symbol 
+                // as the destination for pulling members into
+                var documentWithTypeDeclaration = solutionWithUpdatedOriginalType.GetRequiredDocument(updatedDocument.Id);
+                newType = await GetNewTypeSymbolAsync(documentWithTypeDeclaration, typeAnnotation, cancellationToken).ConfigureAwait(false);
```

Comment:
consider not replacing this local in place.  i prefer a fresh var representing this new transformed stated.  makes it easier for debugging too i find :)